pkg/oidc/provider/configuration.go

```diff
@@ -0,0 +1,59 @@
+package provider
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/rancher/rancher/pkg/settings"
+)
+
+// OpenIDConfiguration represents response from the .well-known/openid-configuration endoind
+type OpenIDConfiguration struct {
+	// Issuer the OIDC provider url
+	Issuer string `json:"issuer"`
+	// AuthorizationEndpoint is the authorization endpoint
+	AuthorizationEndpoint string `json:"authorization_endpoint"`
+	// TokenEndpoint is the token endpoint
+	TokenEndpoint string `json:"token_endpoint"`
+	// UserInfoEndpoint is the userinfo endpoint
+	UserInfoEndpoint string `json:"userinfo_endpoint"`
+	// JWKSURI is the jwksuri endpoint
+	JWKSURI string `json:"jwks_uri"`
+	// ResponseTypesSupported response types supported, only 'code' is supported
+	ResponseTypesSupported []string `json:"response_types_supported"`
+	// SubjectTypesSupported subject types supported, only 'public' is supported
+	SubjectTypesSupported []string `json:"subject_types_supported"`
+	// IDTokenSigningAlgsValuesSupported only RS256 is supported
+	IDTokenSigningAlgsValuesSupported []string `json:"id_token_signing_alg_values_supported"`
+	// CodeChallengeMethodsSupported only S256 is supported
+	CodeChallengeMethodsSupported []string `json:"code_challenge_methods_supported"`
+	// ScopesSupported can be openid, profile, offline_token
+	ScopesSupported []string `json:"scopes_supported"`
+	// GrantTypesSupported only code is supported
+	GrantTypesSupported []string `json:"grant_types_supported"`
```

Comment:
Not sure about the naming, should we use a prefix instead?

I.e.: `supported_response_types`, `supported_subject_types`, ...
```go
type OpenIDConfiguration struct {
	SupportedResponseTypes	[]string `json:"supported_response_types"`
	SupportedSubjectTypes	[]string `json:"supported_subject_types"`
}
```

Reply:
Nevermind, I've just seen that this is the RFC naming. :smile: